airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -0,0 +1,737 @@
+        next_token: Optional[str] = DEFAULT_PAGINATION_TOKEN,
+        verbose: Optional[bool] = False,
+        conn_id: Optional[str] = CONN_ID,
+        region: Optional[str] = REGION,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.clusterName = cluster_name
+        self.maxResults = max_results
+        self.nextToken = next_token
+        self.verbose = verbose
+        self.conn_id = conn_id
+        self.region = region
+
+    def execute(self, context):
+        eks_hook = EKSHook(
+            aws_conn_id=self.conn_id,
+            region_name=self.region,
+        )
+
+        return eks_hook.list_nodegroups(
+            clusterName=self.clusterName,
+            verbose=self.verbose,
+            maxResults=self.maxResults,
+            nextToken=self.nextToken,
+        )
+
+
+class EKSPodOperator(KubernetesPodOperator):
+    """
+    Executes a task in a Kubernetes pod on the specified Amazon EKS Cluster.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:EKSPodOperator`
+
+    :param cluster_name: The name of the Amazon EKS Cluster to execute the task on.
+    :type cluster_name: str
+    :param cluster_role_arn: The Amazon Resource Name (ARN) of the IAM role that provides permissions
+       for the Kubernetes control plane to make calls to AWS API operations on your behalf.
+    :type cluster_role_arn: str
+    :param kube_config_file_path: Path to save the generated kube_config file to.
+    :type kube_config_file_path: str
+    :param in_cluster: If True, look for config inside the cluster; if False look for a local file path.
+    :type in_cluster: bool
+    :param namespace: The namespace in which to execute the pod.
+    :type namespace: str
+    :param pod_context: The security context to use while executing the pod.
+    :type pod_context: str
+    :param pod_name: The unique name to give the pod.
+    :type pod_name: str
+    :param pod_username: The username to use while executing the pod.
+    :type pod_username: str
+    :param aws_profile: The named profile containing the credentials for the AWS CLI tool to use.
+    :param aws_profile: str
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+         If this is None or empty then the default boto3 behaviour is used. If
+         running Airflow in a distributed manner and aws_conn_id is None or
+         empty, then the default boto3 configuration would be used (and must be
+         maintained on each worker node).
+    :type aws_conn_id: str
+
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments,too-many-locals
+        self,
+        cluster_name: str,
+        cluster_role_arn: Optional[str] = None,
+        # A default path will be used if none is provided.
+        kube_config_file_path: Optional[str] = os.environ.get(KUBE_CONFIG_ENV_VAR, DEFAULT_KUBE_CONFIG_PATH),
```

Comment:
What happens if several workers are running multiple pods? Will these credentials be overwritten? It looks like a race condition bug.

Reply:
As it is written, the credential file will be overwritten each pod, yes.  I will look into alternatives.

Reply:
Switched to using a named temp file and tested using a DAG which executes six pods over two clusters concurrently without issue.   I know that's a small test, but it didn't have any issues.

Reply:
This should be addressed in https://github.com/apache/airflow/pull/16571/commits/1c9c54328c30794c824282860c7c5b11dfceca22